src/jit/codegenlinear.h

```diff
@@ -219,6 +223,15 @@ bool genIsRegCandidateLocal(GenTreePtr tree)
     return (varDsc->lvIsRegCandidate());
 }
 
+#ifdef FEATURE_PUT_STRUCT_ARG_STK
+#ifdef _TARGET_X86_
+bool m_pushStkArg;
+#else // !_TARGET_X86_
+unsigned m_stkArgVarNum;
+#endif // !_TARGET_X86_
+unsigned m_stkArgOffset;
+#endif // !FEATURE_PUT_STRUCT_ARG_STK
```

Comment:
This state is needed only for passing structs and changes for each struct that is being passed. 

IMO, a better approach would be to define a struct PutArgStk with the above fields and genPutArgStk() creates a local of such a struct, initializes  and passes it by reference to genPutStructArgStk(), which in turn could pass that param to other as required.